diff --git a/nm-applet-dbus-126.conf b/nm-applet-dbus-126.conf
index 2bd6b1a..fb3904b 100644
--- a/nm-applet-dbus-126.conf
+++ b/nm-applet-dbus-126.conf
@@ -32,6 +32,15 @@
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
                        send_interface="org.freedesktop.NetworkManagerSettings.Connection"/>
         </policy>
+        <policy group="netdev">
+                <allow own="org.freedesktop.NetworkManagerUserSettings"/>
+
+                <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
+                       send_interface="org.freedesktop.NetworkManagerSettings"/>
+
+                <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
+                       send_interface="org.freedesktop.NetworkManagerSettings.Connection"/>
+        </policy>
         <policy context="default">
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
                        send_interface="org.freedesktop.DBus.Introspectable"/>
