Index: network-manager-applet/nm-applet.conf
===================================================================
--- network-manager-applet.orig/nm-applet.conf	2008-06-08 23:36:33.000000000 +0200
+++ network-manager-applet/nm-applet.conf	2008-06-11 12:44:22.000000000 +0200
@@ -20,6 +20,15 @@
 		<!-- Only root can get secrets -->
 		<deny send_interface="org.freedesktop.NetworkManagerSettings.Secrets"/>
 	</policy>
+	<policy group="netdev">
+		<allow own="org.freedesktop.NetworkManagerUserSettings"/>
+
+		<allow send_destination="org.freedesktop.NetworkManagerUserSettings"/>
+		<allow send_interface="org.freedesktop.NetworkManagerSettings"/>
+
+		<!-- Only root can get secrets -->
+		<deny send_interface="org.freedesktop.NetworkManagerSettings.Secrets"/>
+	</policy>
 	<policy context="default">
 		<deny own="org.freedesktop.NetworkManagerUserSettings"/>
 
