Index: network-manager-applet/src/libnm-gtk/Makefile.am
===================================================================
--- network-manager-applet.orig/src/libnm-gtk/Makefile.am	2011-11-03 22:26:22.175793827 +0100
+++ network-manager-applet/src/libnm-gtk/Makefile.am	2011-11-03 22:27:24.031016202 +0100
@@ -38,9 +38,11 @@
 	$(top_builddir)/src/marshallers/libmarshallers.la \
 	$(top_builddir)/src/wireless-security/libwireless-security.la
 
-libnm_gtk_la_LDFLAGS = -Wl,-no-undefined
+libnm_gtk_la_LDFLAGS = -Wl,-no-undefined \
+	-export-symbols-regex '^nma_.*'
+
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libnm-gtk.pc
 
-EXTRA_DIST = libnm-gtk.pc.in $(ui_DATA)
\ No newline at end of file
+EXTRA_DIST = libnm-gtk.pc.in $(ui_DATA)
Index: network-manager-applet/src/applet-agent.c
===================================================================
--- network-manager-applet.orig/src/applet-agent.c	2011-09-16 03:03:35.895836999 +0200
+++ network-manager-applet/src/applet-agent.c	2011-11-03 22:28:17.970338092 +0100
@@ -863,7 +863,7 @@
 		              G_SIGNAL_RUN_FIRST,
 		              G_STRUCT_OFFSET (AppletAgentClass, get_secrets),
 		              NULL, NULL,
-		              nma_marshal_VOID__POINTER_POINTER_STRING_POINTER_UINT_POINTER_POINTER,
+		              _nma_marshal_VOID__POINTER_POINTER_STRING_POINTER_UINT_POINTER_POINTER,
 		              G_TYPE_NONE, 7,
 		              G_TYPE_POINTER, G_TYPE_POINTER, G_TYPE_STRING, G_TYPE_POINTER, G_TYPE_UINT, G_TYPE_POINTER, G_TYPE_POINTER);
 
Index: network-manager-applet/src/applet-device-cdma.c
===================================================================
--- network-manager-applet.orig/src/applet-device-cdma.c	2011-11-03 22:26:22.167793927 +0100
+++ network-manager-applet/src/applet-device-cdma.c	2011-11-03 22:28:48.605952950 +0100
@@ -974,7 +974,7 @@
 	g_object_set_data_full (G_OBJECT (modem), "devinfo", info, cdma_device_info_free);
 
 	/* Registration state change signal */
-	dbus_g_object_register_marshaller (nma_marshal_VOID__UINT_UINT,
+	dbus_g_object_register_marshaller (_nma_marshal_VOID__UINT_UINT,
 	                                   G_TYPE_NONE,
 	                                   G_TYPE_UINT, G_TYPE_UINT, G_TYPE_INVALID);
 	dbus_g_proxy_add_signal (info->cdma_proxy, "RegistrationStateChanged",
@@ -990,7 +990,7 @@
 	                             G_CALLBACK (signal_quality_changed_cb), info, NULL);
 
 	/* Modem property change signal */
-	dbus_g_object_register_marshaller (nma_marshal_VOID__STRING_BOXED,
+	dbus_g_object_register_marshaller (_nma_marshal_VOID__STRING_BOXED,
 	                                   G_TYPE_NONE, G_TYPE_STRING, DBUS_TYPE_G_MAP_OF_VARIANT,
 	                                   G_TYPE_INVALID);
 	dbus_g_proxy_add_signal (info->props_proxy, "MmPropertiesChanged",
Index: network-manager-applet/src/applet-device-gsm.c
===================================================================
--- network-manager-applet.orig/src/applet-device-gsm.c	2011-11-03 22:26:22.167793927 +0100
+++ network-manager-applet/src/applet-device-gsm.c	2011-11-03 22:29:06.457728524 +0100
@@ -1675,7 +1675,7 @@
 	g_object_set_data_full (G_OBJECT (modem), "devinfo", info, gsm_device_info_free);
 
 	/* Registration info signal */
-	dbus_g_object_register_marshaller (nma_marshal_VOID__UINT_STRING_STRING,
+	dbus_g_object_register_marshaller (_nma_marshal_VOID__UINT_STRING_STRING,
 	                                   G_TYPE_NONE,
 	                                   G_TYPE_UINT, G_TYPE_STRING, G_TYPE_STRING, G_TYPE_INVALID);
 	dbus_g_proxy_add_signal (info->net_proxy, "RegistrationInfo",
@@ -1691,7 +1691,7 @@
 	                             G_CALLBACK (signal_quality_changed_cb), info, NULL);
 
 	/* Modem property change signal */
-	dbus_g_object_register_marshaller (nma_marshal_VOID__STRING_BOXED,
+	dbus_g_object_register_marshaller (_nma_marshal_VOID__STRING_BOXED,
 	                                   G_TYPE_NONE, G_TYPE_STRING, DBUS_TYPE_G_MAP_OF_VARIANT,
 	                                   G_TYPE_INVALID);
 	dbus_g_proxy_add_signal (info->props_proxy, "MmPropertiesChanged",
Index: network-manager-applet/src/marshallers/Makefile.am
===================================================================
--- network-manager-applet.orig/src/marshallers/Makefile.am	2011-10-29 01:03:22.297745661 +0200
+++ network-manager-applet/src/marshallers/Makefile.am	2011-11-03 22:27:44.146763313 +0100
@@ -18,10 +18,10 @@
 CLEANFILES = $(BUILT_SOURCES)
 
 nma-marshal.h: nma-marshal.list
-	$(GLIB_GENMARSHAL) $< --prefix=nma_marshal --header > $@
+	$(GLIB_GENMARSHAL) $< --prefix=_nma_marshal --header > $@
 
 nma-marshal.c: nma-marshal.list
-	$(GLIB_GENMARSHAL) $< --prefix=nma_marshal --body > $@
+	$(GLIB_GENMARSHAL) $< --prefix=_nma_marshal --body > $@
 
 nma-marshal-main.c: nma-marshal.c nma-marshal.h
 
Index: network-manager-applet/src/connection-editor/nm-connection-editor.c
===================================================================
--- network-manager-applet.orig/src/connection-editor/nm-connection-editor.c	2011-09-16 03:03:36.067834837 +0200
+++ network-manager-applet/src/connection-editor/nm-connection-editor.c	2011-11-03 22:30:14.108878034 +0100
@@ -372,7 +372,7 @@
 		              G_SIGNAL_RUN_FIRST,
 		              G_STRUCT_OFFSET (NMConnectionEditorClass, done),
 		              NULL, NULL,
-		              nma_marshal_VOID__INT_POINTER,
+		              _nma_marshal_VOID__INT_POINTER,
 		              G_TYPE_NONE, 2, G_TYPE_INT, G_TYPE_POINTER);
 }
 
Index: network-manager-applet/src/gnome-bluetooth/bt-widget.c
===================================================================
--- network-manager-applet.orig/src/gnome-bluetooth/bt-widget.c	2011-11-03 22:26:22.175793827 +0100
+++ network-manager-applet/src/gnome-bluetooth/bt-widget.c	2011-11-03 22:29:56.081104671 +0100
@@ -819,7 +819,7 @@
 		g_message ("%s: calling Connect...", __func__);
 
 		/* Watch for BT device property changes */
-		dbus_g_object_register_marshaller (nma_marshal_VOID__STRING_BOXED,
+		dbus_g_object_register_marshaller (_nma_marshal_VOID__STRING_BOXED,
 		                                   G_TYPE_NONE,
 		                                   G_TYPE_STRING, G_TYPE_VALUE,
 		                                   G_TYPE_INVALID);
