Index: network-manager-applet-0.9.4.1/src/utils/utils.c
===================================================================
--- network-manager-applet-0.9.4.1.orig/src/utils/utils.c	2012-03-24 22:28:06.000000000 +0100
+++ network-manager-applet-0.9.4.1/src/utils/utils.c	2012-03-24 22:37:35.438409576 +0100
@@ -374,7 +374,7 @@
 	                                     text1);
 
 	if (text2)
-		gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (err_dialog), text2);
+		gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (err_dialog), "%s", text2);
 	if (title)
 		gtk_window_set_title (GTK_WINDOW (err_dialog), title);
 
