Index: src/gconf-helpers/nma-gconf-connection.c
===================================================================
--- network-manager-applet/src/gconf-helpers/nma-gconf-connection.c	(revision 953)
+++ network-manager-applet/src/gconf-helpers/nma-gconf-connection.c	(working copy)
@@ -469,7 +469,7 @@
 	object_class->finalize     = finalize;
 
 	connection_class->get_settings = get_settings;
-	connection_class->get_secrets  = get_secrets;
+	connection_class->service_get_secrets  = get_secrets;
 	connection_class->update       = update;
 	connection_class->delete       = delete;
 
