Index: network-manager-applet-0.6.6/nm-applet.desktop
===================================================================
--- network-manager-applet-0.6.6.orig/nm-applet.desktop	2008-11-28 12:54:30.000000000 +0100
+++ network-manager-applet-0.6.6/nm-applet.desktop	2008-11-28 12:55:00.000000000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 Categories=
-OnlyShowIn=GNOME;
+NotShowIn=KDE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=NetworkManager
 X-GNOME-Bugzilla-Component=general
