Index: network-manager-applet/nm-applet.desktop.in
===================================================================
--- network-manager-applet.orig/nm-applet.desktop.in	2011-04-09 07:02:58.000000000 +0200
+++ network-manager-applet/nm-applet.desktop.in	2011-04-09 07:25:39.898695018 +0200
@@ -6,6 +6,7 @@
 Terminal=false
 Type=Application
 NoDisplay=true
+NotShowIn=KDE
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=NetworkManager
 X-GNOME-Bugzilla-Component=general
