---
 src/applet.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

Index: network-manager-applet-0.6.6~rc2/src/applet.c
===================================================================
--- network-manager-applet-0.6.6~rc2.orig/src/applet.c
+++ network-manager-applet-0.6.6~rc2/src/applet.c
@@ -857,19 +857,18 @@
 
 
 static GdkPixbuf *nma_get_connected_icon (NMApplet *applet, NetworkDevice *dev)
 {
 	int strength = 0;
 	GdkPixbuf *pixbuf = NULL;
 
 	g_return_val_if_fail (applet != NULL, NULL);
-	g_return_val_if_fail (dev != NULL, NULL);
 
-	if (network_device_is_wireless (dev))
+	if (dev && network_device_is_wireless (dev))
 	{
 		if (applet->is_adhoc)
 			pixbuf = applet->adhoc_icon;
 		else
 		{
 			strength = CLAMP ((int)network_device_get_strength (dev), 0, 100);
 			if (strength > 80)
 				pixbuf = applet->wireless_100_icon;
@@ -1092,17 +1091,20 @@
 			break;
 
 		case NM_STATE_DISCONNECTED:
 			pixbuf = applet->no_connection_icon;
 			tip = g_strdup (_("No network connection"));
 			break;
 
 		case NM_STATE_CONNECTED:
-			if (network_device_is_wired (act_dev)) {
+			if (! act_dev)
+				tip = g_strdup (_("Manual network configuration"));
+			else if (network_device_is_wired (act_dev))
+			{
 				const char *str_addr = network_device_get_ip4_address (act_dev);
 
 				if (str_addr && !strncmp (str_addr, LL_ADDR_PREFIX, strlen (LL_ADDR_PREFIX)))
 					tip = g_strdup (_("Wired network connection with a self-assigned address"));
 				else
 					tip = g_strdup (_("Wired network connection"));
 			} else if (network_device_is_wireless (act_dev)) {
 				if (applet->is_adhoc)
