---
 src/applet.c    |    4 +++-
 src/nm-device.c |    2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

Index: network-manager-applet-0.6.6/src/applet.c
===================================================================
--- network-manager-applet-0.6.6.orig/src/applet.c
+++ network-manager-applet-0.6.6/src/applet.c
@@ -903,23 +903,25 @@
 				pixbuf = applet->wireless_75_icon;
 			else if (strength > 30)
 				pixbuf = applet->wireless_50_icon;
 			else if (strength > 5)
 				pixbuf = applet->wireless_25_icon;
 			else
 				pixbuf = applet->wireless_00_icon;
 		}
-	} else {
+	} else if (dev) {
 		const char *str_addr = network_device_get_ip4_address (dev);
 
 		if (str_addr && !strncmp (str_addr, LL_ADDR_PREFIX, strlen (LL_ADDR_PREFIX)))
 			pixbuf = applet->wired_autoip_icon;
 		else
 			pixbuf = applet->wired_icon;
+	} else {
+		pixbuf = applet->wired_icon; 
 	}
 
 	return pixbuf;
 }
 
 
 static GdkPixbuf * nma_act_stage_to_pixbuf (NMApplet *applet, NetworkDevice *dev, WirelessNetwork *net, char **tip)
 {
Index: network-manager-applet-0.6.6/src/nm-device.c
===================================================================
--- network-manager-applet-0.6.6.orig/src/nm-device.c
+++ network-manager-applet-0.6.6/src/nm-device.c
@@ -513,17 +513,17 @@
 {
 	g_return_val_if_fail (dev != NULL, NULL);
 
 	return (dev->ip4addr);
 }
 
 void network_device_set_ip4_address (NetworkDevice *dev, const char *addr)
 {
-	g_return_if_fail (dev != NULL);
+	if (!dev) return;
 
 	if (dev->ip4addr)
 		g_free (dev->ip4addr);
 	dev->ip4addr = addr ? g_strdup (addr) : NULL;
 }
 
 /*
  * Accessors for default route
