Index: network-manager-applet/src/applet-device-wifi.c
===================================================================
--- network-manager-applet.orig/src/applet-device-wifi.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/applet-device-wifi.c	2013-01-06 10:33:27.826114129 +0100
@@ -460,6 +460,23 @@
 		nm_connection_add_setting (connection, NM_SETTING (s_8021x));
 	}
 
+	if (utils_default_to_private_connection (applet->nm_client)) {
+		if (!s_con) {
+			s_con = (NMSettingConnection *) nm_setting_connection_new ();
+			nm_connection_add_setting (connection, NM_SETTING (s_con));
+		}
+		nm_setting_connection_add_permission (s_con, "user", g_get_user_name (), NULL);
+
+		if ((rsn_flags & NM_802_11_AP_SEC_KEY_MGMT_PSK) ||
+		    (wpa_flags & NM_802_11_AP_SEC_KEY_MGMT_PSK)) {
+			if (!s_wsec) {
+				s_wsec = (NMSettingWirelessSecurity *) nm_setting_wireless_security_new ();
+				nm_connection_add_setting (connection, NM_SETTING (s_wsec));
+			}
+			g_object_set (s_wsec, NM_SETTING_WIRELESS_SECURITY_PSK_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED, NULL);
+		}
+	}
+
 	/* If it's an 802.1x connection, we need more information, so pop up the
 	 * Dialog Of Doom.
 	 */
Index: network-manager-applet/src/applet-device-gsm.c
===================================================================
--- network-manager-applet.orig/src/applet-device-gsm.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/applet-device-gsm.c	2013-01-06 10:33:27.826114129 +0100
@@ -148,6 +148,7 @@
 		              NM_SETTING_GSM_NUMBER, "*99#",
 		              NM_SETTING_GSM_USERNAME, method->username,
 		              NM_SETTING_GSM_PASSWORD, method->password,
+		              NM_SETTING_GSM_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
 		              NM_SETTING_GSM_APN, method->gsm_apn,
 		              NULL);
 		nm_connection_add_setting (connection, setting);
@@ -175,6 +176,7 @@
 		              NULL);
 		g_free (uuid);
 		g_free (id);
+		nm_setting_connection_add_permission ((NMSettingConnection *) setting, "user", g_get_user_name (), NULL);
 		nm_connection_add_setting (connection, setting);
 	}
 
Index: network-manager-applet/src/applet-device-cdma.c
===================================================================
--- network-manager-applet.orig/src/applet-device-cdma.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/applet-device-cdma.c	2013-01-06 10:33:27.826114129 +0100
@@ -119,6 +119,7 @@
 		              NM_SETTING_CDMA_NUMBER, "#777",
 		              NM_SETTING_CDMA_USERNAME, method->username,
 		              NM_SETTING_CDMA_PASSWORD, method->password,
+		              NM_SETTING_CDMA_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
 		              NULL);
 		nm_connection_add_setting (connection, setting);
 
@@ -145,6 +146,7 @@
 		              NULL);
 		g_free (uuid);
 		g_free (id);
+		nm_setting_connection_add_permission ((NMSettingConnection *) setting, "user", g_get_user_name (), NULL);
 		nm_connection_add_setting (connection, setting);
 	}
 
Index: network-manager-applet/src/gnome-bluetooth/nma-bt-device.c
===================================================================
--- network-manager-applet.orig/src/gnome-bluetooth/nma-bt-device.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/gnome-bluetooth/nma-bt-device.c	2013-01-06 10:37:54.888925216 +0100
@@ -336,6 +336,7 @@
 	              NM_SETTING_CDMA_NUMBER, "#777",
 	              NM_SETTING_CDMA_USERNAME, method->username,
 	              NM_SETTING_CDMA_PASSWORD, method->password,
+	              NM_SETTING_CDMA_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
 	              NULL);
 	nm_connection_add_setting (connection, setting);
 
@@ -362,6 +363,7 @@
 	              NULL);
 	g_free (uuid);
 	g_free (id);
+	nm_setting_connection_add_permission ((NMSettingConnection *) setting, "user", g_get_user_name (), NULL);
 	nm_connection_add_setting (connection, setting);
 
 	return connection;
@@ -381,6 +383,7 @@
 	              NM_SETTING_GSM_NUMBER, "*99#",
 	              NM_SETTING_GSM_USERNAME, method->username,
 	              NM_SETTING_GSM_PASSWORD, method->password,
+	              NM_SETTING_GSM_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
 	              NM_SETTING_GSM_APN, method->gsm_apn,
 	              NULL);
 	nm_connection_add_setting (connection, setting);
@@ -408,6 +411,7 @@
 	              NULL);
 	g_free (uuid);
 	g_free (id);
+	nm_setting_connection_add_permission ((NMSettingConnection *) setting, "user", g_get_user_name (), NULL);
 	nm_connection_add_setting (connection, setting);
 
 	return connection;
@@ -859,6 +863,7 @@
 	              NULL);
 	g_free (id);
 	g_free (uuid);
+	nm_setting_connection_add_permission ((NMSettingConnection *) setting, "user", g_get_user_name (), NULL);
 	nm_connection_add_setting (connection, setting);
 
 	/* The Bluetooth settings */
Index: network-manager-applet/src/utils/utils.c
===================================================================
--- network-manager-applet.orig/src/utils/utils.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/utils/utils.c	2013-01-06 10:33:27.830114171 +0100
@@ -360,3 +360,10 @@
 	}
 }
 
+gboolean
+utils_default_to_private_connection (NMClient *client)
+{
+	NMClientPermissionResult perms;
+	perms = nm_client_get_permission_result (client, NM_CLIENT_PERMISSION_SETTINGS_MODIFY_SYSTEM);
+	return (perms != NM_CLIENT_PERMISSION_RESULT_YES);
+}
Index: network-manager-applet/src/utils/utils.h
===================================================================
--- network-manager-applet.orig/src/utils/utils.h	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/utils/utils.h	2013-01-06 10:33:27.830114171 +0100
@@ -27,6 +27,7 @@
 #include <gtk/gtk.h>
 #include <nm-connection.h>
 #include <nm-device.h>
+#include <nm-client.h>
 #include <net/ethernet.h>
 #include <nm-access-point.h>
 
@@ -55,6 +56,8 @@
                               gboolean modal,
                               GtkWindow *parent);
 
+gboolean utils_default_to_private_connection (NMClient *client);
+
 #define NMA_ERROR (g_quark_from_static_string ("nma-error-quark"))
 
 typedef enum  {
Index: network-manager-applet/src/connection-editor/nm-connection-list.c
===================================================================
--- network-manager-applet.orig/src/connection-editor/nm-connection-list.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/nm-connection-list.c	2013-01-06 10:33:27.830114171 +0100
@@ -605,6 +605,7 @@
 	info->new_func (GTK_WINDOW (list->dialog),
 	                really_add_connection,
 	                page_get_connections,
+	                info->list->nm_client,
 	                info);
 }
 
@@ -1577,6 +1578,7 @@
 		info->new_func (GTK_WINDOW (info->list->dialog),
 		                really_add_connection,
 		                page_get_connections,
+		                info->list->nm_client,
 		                info);
 	}
 }
Index: network-manager-applet/src/wireless-security/ws-wpa-psk.c
===================================================================
--- network-manager-applet.orig/src/wireless-security/ws-wpa-psk.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/wireless-security/ws-wpa-psk.c	2013-01-06 10:33:27.830114171 +0100
@@ -92,11 +92,13 @@
 {
 	GtkWidget *widget;
 	const char *key;
+	NMSettingConnection *s_con;
 	NMSettingWireless *s_wireless;
 	NMSettingWirelessSecurity *s_wireless_sec;
 	const char *mode;
 	gboolean is_adhoc = FALSE;
 
+	s_con = nm_connection_get_setting_connection (connection);
 	s_wireless = nm_connection_get_setting_wireless (connection);
 	g_assert (s_wireless);
 
@@ -113,6 +115,9 @@
 	widget = GTK_WIDGET (gtk_builder_get_object (parent->builder, "wpa_psk_entry"));
 	key = gtk_entry_get_text (GTK_ENTRY (widget));
 	g_object_set (s_wireless_sec, NM_SETTING_WIRELESS_SECURITY_PSK, key, NULL);
+	if (s_con && nm_setting_connection_get_num_permissions (s_con)) {
+		g_object_set (s_wireless_sec, NM_SETTING_WIRELESS_SECURITY_PSK_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED, NULL);
+	}
 
 	wireless_security_clear_ciphers (connection);
 	if (is_adhoc) {
Index: network-manager-applet/src/connection-editor/page-mobile.c
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-mobile.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-mobile.c	2013-01-06 10:33:27.830114171 +0100
@@ -554,6 +554,7 @@
 	NMConnection *connection = NULL;
 
 	if (!canceled && method) {
+		NMSettingConnection *s_con;
 		NMSetting *type_setting;
 		const char *ctype = NULL;
 		char *detail = NULL;
@@ -567,6 +568,7 @@
 			              NM_SETTING_GSM_NUMBER, "*99#",
 			              NM_SETTING_GSM_USERNAME, method->username,
 			              NM_SETTING_GSM_PASSWORD, method->password,
+			              NM_SETTING_GSM_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
 			              NM_SETTING_GSM_APN, method->gsm_apn,
 			              NULL);
 			break;
@@ -578,6 +580,7 @@
 			              NM_SETTING_CDMA_NUMBER, "#777",
 			              NM_SETTING_GSM_USERNAME, method->username,
 			              NM_SETTING_GSM_PASSWORD, method->password,
+			              NM_SETTING_GSM_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
 			              NULL);
 			break;
 		default:
@@ -592,6 +595,13 @@
 		connection = ce_page_new_connection (detail, ctype, FALSE, info->get_connections_func, info->user_data);
 		g_free (detail);
 
+		s_con = nm_connection_get_setting_connection (connection);
+		if (!s_con) {
+			s_con = (NMSettingConnection *) nm_setting_connection_new ();
+			nm_connection_add_setting (connection, NM_SETTING (s_con));
+		}
+		nm_setting_connection_add_permission (s_con, "user", g_get_user_name (), NULL);
+
 		nm_connection_add_setting (connection, type_setting);
 		add_default_serial_setting (connection);
 		nm_connection_add_setting (connection, nm_setting_ppp_new ());
@@ -614,6 +624,7 @@
 mobile_connection_new (GtkWindow *parent,
                        PageNewConnectionResultFunc result_func,
                        PageGetConnectionsFunc get_connections_func,
+                       NMClient *client,
                        gpointer user_data)
 {
 	NMAMobileWizard *wizard;
Index: network-manager-applet/src/connection-editor/page-vpn.c
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-vpn.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-vpn.c	2013-01-06 10:33:27.834114213 +0100
@@ -186,10 +186,12 @@
 vpn_connection_new (GtkWindow *parent,
                     PageNewConnectionResultFunc result_func,
                     PageGetConnectionsFunc get_connections_func,
+                    NMClient *client,
                     gpointer user_data)
 {
 	char *service = NULL;
 	NMConnection *connection;
+	NMSettingConnection *s_con;
 	NMSetting *s_vpn;
 
 	service = vpn_ask_connection_type (parent);
@@ -203,6 +205,14 @@
 	                                     FALSE,
 	                                     get_connections_func,
 	                                     user_data);
+
+	s_con = nm_connection_get_setting_connection (connection);
+	if (!s_con) {
+		s_con = (NMSettingConnection *) nm_setting_connection_new ();
+		nm_connection_add_setting (connection, NM_SETTING (s_con));
+	}
+	nm_setting_connection_add_permission (s_con, "user", g_get_user_name (), NULL);
+
 	s_vpn = nm_setting_vpn_new ();
 	g_object_set (s_vpn, NM_SETTING_VPN_SERVICE_TYPE, service, NULL);
 	g_free (service);
Index: network-manager-applet/src/connection-editor/ce-page.h
===================================================================
--- network-manager-applet.orig/src/connection-editor/ce-page.h	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/ce-page.h	2013-01-06 10:33:27.834114213 +0100
@@ -43,6 +43,7 @@
 typedef void (*PageNewConnectionFunc) (GtkWindow *parent,
                                        PageNewConnectionResultFunc result_func,
                                        PageGetConnectionsFunc get_connections_func,
+                                       NMClient *client,
                                        gpointer user_data);
 
 #define CE_TYPE_PAGE            (ce_page_get_type ())
Index: network-manager-applet/src/connection-editor/page-dsl.c
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-dsl.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-dsl.c	2013-01-06 10:33:27.834114213 +0100
@@ -224,6 +224,7 @@
 dsl_connection_new (GtkWindow *parent,
                     PageNewConnectionResultFunc result_func,
                     PageGetConnectionsFunc get_connections_func,
+                    NMClient *client,
                     gpointer user_data)
 {
 	NMConnection *connection;
Index: network-manager-applet/src/connection-editor/page-dsl.h
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-dsl.h	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-dsl.h	2013-01-06 10:33:27.834114213 +0100
@@ -56,6 +56,7 @@
 void dsl_connection_new (GtkWindow *parent,
                          PageNewConnectionResultFunc callback,
                          PageGetConnectionsFunc get_connections_func,
+                         NMClient *client,
                          gpointer user_data);
 
 #endif  /* __PAGE_DSL_H__ */
Index: network-manager-applet/src/connection-editor/page-mobile.h
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-mobile.h	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-mobile.h	2013-01-06 10:33:27.834114213 +0100
@@ -56,6 +56,7 @@
 void mobile_connection_new (GtkWindow *parent,
                             PageNewConnectionResultFunc result_func,
                             PageGetConnectionsFunc get_connections_func,
+                            NMClient *client,
                             gpointer user_data);
 
 #endif  /* __PAGE_MOBILE_H__ */
Index: network-manager-applet/src/connection-editor/page-vpn.h
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-vpn.h	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-vpn.h	2013-01-06 10:33:27.834114213 +0100
@@ -56,6 +56,7 @@
 void vpn_connection_new (GtkWindow *parent,
                          PageNewConnectionResultFunc result_func,
                          PageGetConnectionsFunc get_connections_func,
+                         NMClient *client,
                          gpointer user_data);
 
 #endif  /* __PAGE_VPN_H__ */
Index: network-manager-applet/src/connection-editor/page-wired.c
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-wired.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-wired.c	2013-01-06 10:33:27.834114213 +0100
@@ -441,6 +441,7 @@
 wired_connection_new (GtkWindow *parent,
                       PageNewConnectionResultFunc result_func,
                       PageGetConnectionsFunc get_connections_func,
+                      NMClient *client,
                       gpointer user_data)
 {
 	NMConnection *connection;
Index: network-manager-applet/src/connection-editor/page-wired.h
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-wired.h	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-wired.h	2013-01-06 10:33:27.834114213 +0100
@@ -56,6 +56,7 @@
 void wired_connection_new (GtkWindow *parent,
                            PageNewConnectionResultFunc result_func,
                            PageGetConnectionsFunc get_connections_func,
+                           NMClient *client,
                            gpointer user_data);
 
 #endif  /* __PAGE_WIRED_H__ */
Index: network-manager-applet/src/connection-editor/page-wireless.c
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-wireless.c	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-wireless.c	2013-01-06 10:33:27.834114213 +0100
@@ -34,6 +34,8 @@
 #include <nm-utils.h>
 #include <net/if_arp.h> /* for ARPHRD_ETHER for MAC utilies */
 
+#include "utils.h"
+
 #include "page-wireless.h"
 
 G_DEFINE_TYPE (CEPageWireless, ce_page_wireless, CE_TYPE_PAGE)
@@ -653,6 +655,7 @@
 wifi_connection_new (GtkWindow *parent,
                      PageNewConnectionResultFunc result_func,
                      PageGetConnectionsFunc get_connections_func,
+                     NMClient *client,
                      gpointer user_data)
 {
 	NMConnection *connection;
@@ -663,6 +666,17 @@
 	                                     TRUE,
 	                                     get_connections_func,
 	                                     user_data);
+
+	if (utils_default_to_private_connection (client)) {
+		NMSettingConnection *s_con;
+		s_con = nm_connection_get_setting_connection (connection);
+		if (!s_con) {
+			s_con = (NMSettingConnection *) nm_setting_connection_new ();
+			nm_connection_add_setting (connection, NM_SETTING (s_con));
+		}
+		nm_setting_connection_add_permission (s_con, "user", g_get_user_name (), NULL);
+	}
+
 	s_wifi = nm_setting_wireless_new ();
 	g_object_set (s_wifi, NM_SETTING_WIRELESS_MODE, "infrastructure", NULL);
 	nm_connection_add_setting (connection, s_wifi);
Index: network-manager-applet/src/connection-editor/page-wireless.h
===================================================================
--- network-manager-applet.orig/src/connection-editor/page-wireless.h	2013-01-06 10:33:27.838114255 +0100
+++ network-manager-applet/src/connection-editor/page-wireless.h	2013-01-06 10:33:27.834114213 +0100
@@ -60,6 +60,7 @@
 void wifi_connection_new (GtkWindow *parent,
                           PageNewConnectionResultFunc result_func,
                           PageGetConnectionsFunc get_connections_func,
+                          NMClient *client,
                           gpointer user_data);
 
 #endif  /* __PAGE_WIRELESS_H__ */
Index: network-manager-applet/src/libnm-gtk/nm-wifi-dialog.c
===================================================================
--- network-manager-applet.orig/src/libnm-gtk/nm-wifi-dialog.c	2013-01-06 10:20:23.729832803 +0100
+++ network-manager-applet/src/libnm-gtk/nm-wifi-dialog.c	2013-01-06 10:47:34.871015644 +0100
@@ -1217,6 +1217,9 @@
 			      NM_SETTING_CONNECTION_UUID, uuid,
 			      NULL);
 		g_free (uuid);
+		if (utils_default_to_private_connection (priv->client)) {
+			nm_setting_connection_add_permission (s_con, "user", g_get_user_name (), NULL);
+		}
 		nm_connection_add_setting (connection, (NMSetting *) s_con);
 
 		s_wireless = (NMSettingWireless *) nm_setting_wireless_new ();
